packages/amino/src/encoding.ts

```diff
@@ -77,6 +81,84 @@ export function decodeBech32Pubkey(bechEncoded: string): Pubkey {
   return decodeAminoPubkey(data);
 }
 
+/**
+ * Uvarint decoder for Amino.
+ * Decoding numbers > 127 is not supported here. Please tell those lazy CosmJS maintainers to port the binary.Varint implementation from the Go standard library and write some tests.
+ *
+ * @see https://github.com/tendermint/go-amino/blob/8e779b71f40d175/decoder.go#L64-76
+ * @returns varint as number, and bytes count occupied by varaint
+ */
+function decodeUvarint(reader: number[]): [number, number] {
+  if (reader.length < 1) {
+    throw new Error("Can't decode varint. EOF");
+  }
+  return [reader[0], 1];
```

Comment:
We need to check that `reader[0]` is <= 127. Higher values indicate that more bytes are following.